test/powershell/Language/Scripting/NativeExecution/NativeCommandProcessor.Tests.ps1

```diff
@@ -119,5 +119,108 @@ Describe "Native Command Processor" -tags "Feature" {
+                Remove-Item $HOME/nativeCommandProcessor.Success -Force -ErrorAction SilentlyContinue
+                Set-Content -Path "$appFolder/nativeCommandProcessor.desktop" -Force -Value @"
+[Desktop Entry]
+Version=1.0
+Name=nativeCommandProcessor
+Comment=Validate_native_command_processor_open_text_file
+Exec=/bin/sh -c 'echo %u > ~/nativeCommandProcessor.Success'
+Icon=utilities-terminal
+Terminal=true
+Type=Application
+Categories=Application;
+"@
+                xdg-mime default nativeCommandProcessor.desktop text/plain
+            }
+        }
+        elseif ($IsWindows) {
+            $supportedEnvironment = ![System.Management.Automation.Platform]::IsNanoServer -and ![System.Management.Automation.Platform]::IsIoT
+            if ($supportedEnvironment) {
+                cmd /c assoc .foo=foofile
+                cmd /c ftype foofile=cmd /c echo %1^> $TestDrive\foo.txt
+                Remove-Item $TestDrive\foo.txt -Force -ErrorAction SilentlyContinue
+            }
+        }
+    }
+
+    AfterAll {
+        Remove-Item -Path $TestFile -Force -ErrorAction SilentlyContinue
+
+        if ($IsLinux -and $supportedEnvironment) {
+            xdg-mime default $mimeDefault text/plain
+            Remove-Item $appFolder/nativeCommandProcessor.desktop -Force -ErrorAction SilentlyContinue
+            Remove-Item $HOME/nativeCommandProcessor.Success -Force -ErrorAction SilentlyContinue
+        }
+        elseif ($IsWindows -and $supportedEnvironment) {
+            cmd /c assoc .foo=
+            cmd /c ftype foofile=
+        }
+    }
 
+    It "Should open text file without error" -Skip:(!$supportedEnvironment) {
+        if ($IsOSX) {
+            $expectedTitle = Split-Path $TestFile -Leaf
+            open -F -a TextEdit
+            $beforeCount = [int]('tell application "TextEdit" to count of windows' | osascript)
+            & $TestFile
+            $startTime = Get-Date
+            $title = [String]::Empty
+            while (((Get-Date) - $startTime).TotalSeconds -lt 30 -and ($title -ne $expectedTitle)) {
+                Start-Sleep -Milliseconds 100
+                $title = 'tell application "TextEdit" to get name of front window' | osascript
+            }
+            $afterCount = [int]('tell application "TextEdit" to count of windows' | osascript)
+            $afterCount | Should Be ($beforeCount + 1)
+            $title | Should Be $expectedTitle
+            "tell application ""TextEdit"" to close window ""$expectedTitle""" | osascript
+            'tell application "TextEdit" to quit' | osascript
+        }
+        elseif ($IsLinux) {
+            # Validate on Linux by reassociating default app for text file
+            & $TestFile
+            $startTime = Get-Date
+            # It may take time for handler to start
+            while (((Get-Date) - $startTime).TotalSeconds -lt 10 -and (-not (Test-Path "$HOME/nativeCommandProcessor.Success"))) {
```

Comment:
Polling for a file can be a common pattern. Consider implementing a generic file watcher function in HelpersCommon.psm1